File: test-suite-kotlin/src/test/kotlin/io/micronaut/microstream/docs/CustomerRepository.kt
package io.micronaut.microstream.docs

import javax.validation.Valid
import javax.validation.constraints.NotBlank

interface CustomerRepository {
    fun save(customerSave: @Valid CustomerSave): Customer
    fun update(id: @NotBlank String, customerSave: @Valid CustomerSave)
    fun findById(id: @NotBlank String): Customer?
    fun deleteById(id: @NotBlank String)
}

File: test-suite-python/src/test/python/micronaut/microstream/docs/CustomerRepository.py
from typing import Annotated

from jakarta.validation import Valid
from jakarta.validation.constraints import NotBlank, NotNull

from .Customer import Customer
from .CustomerSave import CustomerSave


class CustomerRepository:

    def save(self, customer_save: Annotated[CustomerSave, NotNull, Valid]) -> Customer:
        ...

    def update(self, id: Annotated[str, NotBlank],
               customer_save: Annotated[CustomerSave, NotNull, Valid]) -> None:
        ...

    def find_by_id(self, id: Annotated[str, NotBlank]) -> Customer | None:
        ...

    def delete_by_id(self, id: Annotated[str, NotBlank]) -> None:
        ...
